dataprofiler/profilers/profile_builder.py

```diff
@@ -1582,8 +1589,21 @@ def __add__(  # type: ignore[override]
             merged_profile.row_is_null_count = (
                 self.row_is_null_count + other.row_is_null_count
             )
-            merged_profile.hashed_row_dict.update(self.hashed_row_dict)
-            merged_profile.hashed_row_dict.update(other.hashed_row_dict)
+
+            if (
+                self.options.row_statistics.unique_count.is_enabled
+                and other.options.row_statistics.unique_count.is_enabled
+            ):
```

Comment:
if they disagree, should we raise an exception instead?

Reply:
the problem becomes it silently continues and still profiles and can give a false report.

Reply:
handled in `add_error_checks`